wagtail/admin/views/pages/delete.py

```diff
@@ -17,32 +18,66 @@ def delete(request, page_id):
         raise PermissionDenied
 
     with transaction.atomic():
-        for fn in hooks.get_hooks('before_delete_page'):
+        for fn in hooks.get_hooks("before_delete_page"):
             result = fn(request, page)
-            if hasattr(result, 'status_code'):
+            if hasattr(result, "status_code"):
                 return result
 
         next_url = get_valid_next_url_from_request(request)
 
-        if request.method == 'POST':
+        pages_to_delete = {page}
+
+        # The `construct_synced_page_tree_list` hook returns translation and
+        # alias pages when the action is set to "delete"
+        if getattr(settings, 'WAGTAIL_I18N_ENABLED', False):
+            for fn in hooks.get_hooks("construct_synced_page_tree_list"):
+                fn_pages = fn([page], "delete")
+                if fn_pages and isinstance(fn_pages, dict):
+                    for additional_pages in fn_pages.values():
+                        pages_to_delete.update(additional_pages)
+
+        pages_to_delete = list(pages_to_delete)
+
+        if request.method == "POST":
             parent_id = page.get_parent().id
+            # Delete the source page.
             action = DeletePageAction(page, user=request.user)
             # Permission checks are done above, so skip them in execute.
             action.execute(skip_permission_checks=True)
 
-            messages.success(request, _("Page '{0}' deleted.").format(page.get_admin_display_title()))
+            # Delete translation and alias pages if they have the same parent page.
+            parent_page_translations = page.get_parent().get_translations()
+            for _page in pages_to_delete:
+                if _page.get_parent() in parent_page_translations:
+                    action = DeletePageAction(_page, user=request.user)
+                    # Permission checks are done above, so skip them in execute.
+                    action.execute(skip_permission_checks=True)
+
+            messages.success(
+                request, _("Page '{0}' deleted.").format(page.get_admin_display_title())
+            )
 
-            for fn in hooks.get_hooks('after_delete_page'):
+            for fn in hooks.get_hooks("after_delete_page"):
```

Comment:
thinking out loud: what happens if the site has a custom `after_delete_page` hook implementation that does something specific (e.g. notifies an admin). 

We call this for the source page as before, which is great. Should we call the hook for the translated pages? cc @phildexter

Reply:
That's a great question. I feel like, yes, we should call the hook for translated pages. Also thinking out loud here: 

Would we want to create a _new_ i18n enabled hook or just pass all the translated pages into the existing hook? If we pass it into the existing `after_delete_page` hook, would that break past code? If we created its own i18n enabled hook, we are creating further divergence in hooks, no? Though if we had a bunch of i18n-only hooks it could make sense to create them now (or soon..ish).

Reply:
Not sure really. I think if I'm having an opinion, if it's an alias page then don't call it (because the page didn't really exist as a normal page) but if it's a translated page then do call it...does that feel consistent with the other things we're doing?

Reply:
Let's call it for all, but only handle redirects for the page the delete was called on.

Reply:
@zerolab @phildexter Redirects are only automatically created for `move` actions and when a slug is changed. I can add that in #8032 though. If we agree to do that I'll add a todo on that PR so we don't forget to add it.

Reply:
redirects in this case was probablu a bit misleading. it refers to 

```python
            for fn in hooks.get_hooks("after_delete_page"):
                result = fn(request, page)
                if hasattr(result, "status_code"):
                    return result
```

i.e. if the hook returns a result that has a status_code (usually a `HttpResponse`), then this returns. 

what I meant is call the after hook for the other pages, but don't return anything

Reply:
Updated here: https://github.com/wagtail/wagtail/pull/8002/commits/5ba40c00b2fcc56345b0d41453d30d561a29c6ea 

Let me know if this _isn't_ what we were thinking.

Reply:
https://github.com/wagtail/wagtail-personalisation/blob/48955675be73ba0753229660aae9c774c6f65bed/src/wagtail_personalisation/wagtail_hooks.py#L304

Might be another relevant case to consider. As well as the fact that wagtail_personalization will prevent pages from being deleted if they are canonical: 

https://github.com/wagtail/wagtail-personalisation/blob/137b5b411ccbe67cca2cc84fe1e7f63c78e8133e/src/wagtail_personalisation/models.py#L215

So I'm curious how those sorts of things would interact with this? 

That being said, I think it's overall a great idea! (They said as they spent plenty of time yesterday manually deleting pages and their translations in the shell the other day...)

Reply:
Hmm... that should be fine as that happens before all this logic